Version 1.0

SEOKMINJUN opened the pull request:
## What
Merge develop to master for release v1.0

SEOKMINJUN committed:
Refactor: Make Managers Singleton and move getInstance to Globals

SEOKMINJUN committed:
Refactor : Fix Managers, Merge AchievementManager with Conditions, Move Settings to Globals
 - Change getManagers working properly
 - Change drawPost to draw on Screen
 - Merge AchievementManager with Conditions
  - Check Achievement on Statistics or Round End.
 - Merge Settings to Globals
 - Move clove's Stuffs to Engine

SEOKMINJUN committed:
Chore : Add Project Lombok to Gradle

SEOKMINJUN committed:
Refactor : Use Setter on spriteType on Entity

SEOKMINJUN committed:
feat : Add draw on Entity

SEOKMINJUN committed:
fix : use draw instead of drawPost

SEOKMINJUN committed:
Refactor : Move SoundManager to engine

SEOKMINJUN committed:
Refactor : Make playES and playBGM static

SEOKMINJUN committed:
Refactor : Move ScoreManager to Entity

SEOKMINJUN committed:
Refactor : Change Entity to EntityBase array getEntities

SEOKMINJUN committed:
Refactor : Add ScoreManager to Globals

SEOKMINJUN committed:
Feat : Add EntityBase

SEOKMINJUN committed:
Refactor : Move PlayerGrowth to Ship

SEOKMINJUN committed:
Refactor : Remove PlayerGrowth import / file

SEOKMINJUN committed:
Refactor : Move class and Use Draw/Update method
 - Move class from team folder to entity/engine
 - Refactor GameScreen Update method/draw to each entity's update/draw method(WIP)

SEOKMINJUN committed:
Chore : add build CI

SEOKMINJUN committed:
Chore : update build CI

SEOKMINJUN committed:
KAN-3 Refactor: Remove getCurrentGameState() on Globals

SEOKMINJUN committed:
KAN-12 Chore: Add CODEOWNERS for auto assign reviewers

SEOKMINJUN committed:
Merge pull request #3 from hy-clove/KAN-12-Chore/CodeOwners

KAN-12 Chore: Add CODEOWNERS for auto assign reviewers

DongjunSuh923 committed:
Merge pull request #2 from hy-clove/refactor

KAN-4 Refactor: GameScreen and Managers

RyuJeMu committed:
update about calculating score

SEOKMINJUN committed:
KAN-20 Refactor: TwoPlayerMode/Screen/Ship
- Remove TwoPlayer stuff at other classes
- Make screen update method return need to draw
- Make draw callable on subclass
- Move health variable from screen to ship entity

SEOKMINJUN committed:
KAN-20 Refactor: ItemManager/ItemPool/EnemyShip/Ship/GameScreen
- Remove ItemManager/ItemPool
 - move spawn item method to GameScreen
 - Use item method at item entity
- Remove obstacle debug log
- Fix ship health at other entity
- Remove dummy method on Ship
- Move dropItem method into
- Use Getter annotation on GameScreen

SEOKMINJUN committed:
KAN-20 KAN-22 Refactor: TwoPlayerMode/Ship/Health Check
- Remove reinitialize GameSettings in Core's TwoPlayerMode
- Add Health Check on GameScreen initialize(For TwoPlayerMode)
- Change health check method on ship
 - Do not draw when ship dead
 - Disable collision test with dead ship
 - rename destory method at ship to playDestoryAnimation
- extract checkGameEnd from update for override at TwoModePlayer class

SEOKMINJUN committed:
KAN-20 Refactor: GameState
- Remove other constructor to prevent the variables from disappearing

SEOKMINJUN committed:
Fix: Ship disappear when damaged

SEOKMINJUN committed:
KAN-20 Refactor: Categorize team classes
- DrawManagerImpl to engine
- RoundState to engine
- TwoPlayerMode to screen and rename it to TwoPlayerGameScreen
- Background to engine

RyuJeMu committed:
Merge pull request #4 from hy-clove/KAN-6-Fix-manageCollisions_add_item-Need-to-update-about-calculating-score

update about calculating score

DongjunSuh923 committed:
-Changed AchievementList name properly
-Added AchievementType 'Accuracy'
-Added Accuracy to gameState
-Added Accuracy to FileManager
-Added method comAccuracy to Statistics.java

SEOKMINJUN committed:
Merge remote-tracking branch 'origin/develop' into KAN-20-refactor

# Conflicts:
#	src/engine/DrawManagerImpl.java
#	src/screen/GameScreen.java

SEOKMINJUN committed:
Fix: fix package on DrawManagerImpl

woojinss committed:
Create: Collections Menu
Creating the Basic Framework for the Collections Menu
Accessible from the Main Menu

woojinss committed:
Modified : Collections Menu Category
The categories in the Collection Menu can be switched using the arrow keys.

SEOKMINJUN committed:
KAN-23 Fix: Sound was not loaded when packed.

SEOKMINJUN committed:
Refactor: Remove unused import

woojinss committed:
Modified: Create CollectionManager and Save Collection Data to a File
For Save Collection Data, I Created CollectionManager file.
And I Temporarily Excluded the Coin Encryption Function.  for Smoother Testing.

SEOKMINJUN committed:
Merge pull request #6 from hy-clove/KAN-20-refactor

KAN-20 Refactor

SEOKMINJUN committed:
Merge pull request #7 from hy-clove/KAN-23-Fix/SoundFileLoad

KAN-23 Fix/Sound file load

HanRH99 committed:
KAN-7 Feat/add player y-movement (both 1 and 2 playermode)

woojinss committed:
Modified : CollectionScreen Enemy Kills
Configured the CollectionsScreen to display the kill count for each EnemyType

woojinss committed:
Merge branch 'refs/heads/develop' into KAN-8-Create-a-compendium-tab-to-record-encountered-enemies-achievements-and-more

# Conflicts:
#	src/entity/ItemManager.java

woojinss committed:
develop and KAN-8 merged

woojinss committed:
Modified : CollectionScreen Item Gain
Configured the CollectionsScreen to display the gain count for each ItemType
Add names for each Enemy and Item type.

woojinss committed:
Modified : CollectionScreen Achievement Clear
Configured the CollectionsScreen to display the clear count for each Achievement Type
Add Achievement's names and descriptions

HanRH99 committed:
Merge pull request #8 from hy-clove/KAN-7-Feat-Add-player-y-movement

KAN-7 Feat/add player y-movement (both 1 and 2 player mode)

SEOKMINJUN committed:
Chore: Add PR reviewer notify

SEOKMINJUN committed:
Merge pull request #9 from hy-clove/KAN-28-Chore/ReviewNotify

Chore: Add PR reviewer notify

DongjunSuh923 committed:
-GameScreen.java : Added maxShipsDestructionStreak, checkAndUpdateStreak / Added several codes related to addShipsDestroyed(Statistics.properties might bug out with out it)
-Globals.java : Added getStatistics for scheduler in the statistics.java
-PiercingBullet.java : onCollision will now call getStatistics.setShipDestroyed
-ScoreScreen.java : Added several codes to saveStatistics
-Statistics.java : Added Scheduler(Won't update immediately if it doesn't exist), Added checkAndUpdateStreak, comShipsDestructionStreak for kill streak achievement

DongjunSuh923 committed:
Merge remote-tracking branch 'origin/develop' into KAN-26-Fix-Add-Accuracy-to-statistics.properties-and-fix-Kill-Streak-system

# Conflicts:
#	src/engine/Core.java
#	src/screen/GameScreen.java

DongjunSuh923 committed:
-Fixed A problem that the game crashes when your ship loses all its life.(Problem occured by parseInt/parseFloat problem)

woojinss committed:
Modified : Reflect Item Gain to Collection
Item Gain is now reflected in the Collection.
Add code comments

SEOKMINJUN committed:
Merge pull request #12 from hy-clove/KAN-8-Create-a-compendium-tab-to-record-encountered-enemies-achievements-and-more

Kan 8 create a compendium tab to record encountered enemies achievements and more

SEOKMINJUN committed:
Merge pull request #11 from hy-clove/KAN-26-Fix-Add-Accuracy-to-statistics.properties-and-fix-Kill-Streak-system

Kan 26 fix add accuracy to statistics.properties and fix kill streak system

SEOKMINJUN committed:
Chore: Add ffsampledsp library to support MP3 file

SEOKMINJUN committed:
Chore: Convert wav file to mp3

SEOKMINJUN committed:
Chore: Make build target from java to application

SEOKMINJUN committed:
Fix: Compile error on AchievementList, CollectionManager
- Add access level at Achievement from AchievementList
- Update Achievements at CollectionManager

SEOKMINJUN committed:
Fix: Compile error on AchievementList, CollectionManager
- Add access level at Achievement from AchievementList
- Update Achievements at CollectionManager

SEOKMINJUN committed:
Refactor: Rename UpgradeManager to ShipUpgradeManager and ShipStatus to ShipUpgradeStatus

SEOKMINJUN committed:
Merge pull request #14 from hy-clove/KAN-32-Chore/UpdateGradle

KAN-32 Chore: Update Gradle

SEOKMINJUN committed:
KAN-34 Fix: Fix statistic
- Add default accuracy value
- Make Statistic.properties thread safe

SEOKMINJUN committed:
KAN-25 KAN-35 Refactor: Update Item(Bomb), Score, EnemyShip
- Move destroy enemyShip from EnemyShipFormation to EnemyShip
- Change explosion algorithm to circle collide test
- Move bomb item to Ship from Bomb class
- Move ItemBarrierAndHeart stuff to Ship
- Move ScoreManager stuff to GameScreen
- Move health system for enemyShip from HpEnemyShip to EnemyShip
- Create BombBullet extends PiercingBullet(For use method from PiercingBullet)
- Bomb can be exploded when collide with Obstacle
- Fix Player 2 can't get some items

SEOKMINJUN committed:
Chore: Remove solved TODO

SEOKMINJUN committed:
Refactor : Remove unused imports, Remove ScoreManager

SEOKMINJUN committed:
Fix : Compile error

SEOKMINJUN committed:
Merge pull request #15 from hy-clove/KAN-33-Fix/AchievementList

KAN-33 Fix Compile error on AchievementList, Update CollectionManager

RyuJeMu committed:
add calculating player-distance

SEOKMINJUN committed:
Fix : Ship can moveable when got damaged

SEOKMINJUN committed:
Fix : Removing EnemyShip from EnemyShipFormation is too late.

SEOKMINJUN committed:
Fix : Add accuracy field to recent file

RyuJeMu committed:
resolve merge conflicts

SEOKMINJUN committed:
Merge pull request #17 from hy-clove/KAN-27-Feat-Distance-traveled-calculation

KAN-27-Feat-Distance-traveled-calculation

SEOKMINJUN committed:
Merge pull request #13 from hy-clove/KAN-24-Feat/MP3

KAN-24 feat : Add support for MP3

SEOKMINJUN committed:
Merge branch 'develop' into KAN-25-Refactor/ItemScore

# Conflicts:
#	src/engine/Achievement/AchievementList.java
#	src/entity/Ship.java
#	src/screen/GameScreen.java

SEOKMINJUN committed:
Fix: Update sound file extension

SEOKMINJUN committed:
Merge pull request #16 from hy-clove/KAN-25-Refactor/ItemScore

KAN-25 KAN-34 KAN-35 Refactor : Item, EnemyShip, Score Logic

RyuJeMu committed:
Update sprite type based on destruction state

RyuJeMu committed:
Teleport to screen edge on right or left double-tap

SEOKMINJUN committed:
Chore : Update build file for test

SEOKMINJUN committed:
Test : Add test for Ship, BombBullet

SEOKMINJUN committed:
Merge pull request #19 from hy-clove/KAN-14-JUnit/ShipEntity

KAN-14 Add Junit test for Ship, BombBullet

RyuJeMu committed:
Added exception handling for edge cases in Double Tap functionality, ensuring robust boundary checks and preventing invalid inputs.

RyuJeMu committed:
Logic not working

woojinss committed:
Create : Add Simple Pause Menu
I Created Simple Pause Menu.
there's no time freeze

woojinss committed:
Modified : Freeze Time
Function added. Freeze Time if turn on the Pause Menu

RyuJeMu committed:
Cooldown and Double-Tap functionality implementation

RyuJeMu committed:
Modify cooldown-circle, add timer which shows cooldown

woojinss committed:
Modified : Add Pause Menu Option
I Created Pause Menu Options with Two Choices: 'Resume' and 'Go to Title'

woojinss committed:
Modified : Add Pause Menu Option Scrolling
I Created Scrolling for the Pause Menu Options.
If You Try to Move Below the Last Option, it Loops Back to the First Option.

RyuJeMu committed:
Add detectedDoubleTab method in Ship class, delete and some changes in GameScreen

RyuJeMu committed:
And result of Distance-Calculation shows on game screen.

RyuJeMu committed:
Fixing a bug where double tap works while other keys are pressed

SEOKMINJUN committed:
Update double tap detect logic.
- Make double tap detecting logic independent with InputManager
- Remove double tap stuff at GameScreen, InputManager

SEOKMINJUN committed:
Chore : Update build.gradle to print test status

woojinss committed:
Modified : Go To Title Function
A feature to return to the title screen has been implemented.
It works for both single-player and two-player modes.

SEOKMINJUN committed:
Add BombBulletTest, ItemTest, ShipTest
Remove EntityTest

woojinss committed:
Modified : Added Confirmation Window
A warning window has been added to confirm if the player really wants to exit to the title screen.

RyuJeMu committed:
Merge pull request #20 from hy-clove/KAN-36-Feat-Double-tab-teleport

Kan 36 feat double tab teleport

SEOKMINJUN committed:
Merge pull request #21 from hy-clove/KAN-14-JUnit/ShipEntity

KAN-14 JUnit add more entity test

SEOKMINJUN committed:
Merge pull request #22 from hy-clove/KAN-38-Add-Pause-Menu

KAN-38 Add pause menu

SEOKMINJUN committed:
Fix: Update sound file loading method
- Get inputstream by classloader
- Wrap InputStream by BufferdInputStream for AudioInputStream (For supporting mark())

SEOKMINJUN committed:
Chore : Make it packaging library

DongjunSuh923 committed:
Achievement.java : added switch-case to checkValue
AchievementList.java : changed achievement name
CollectionManager.java : added getAchievementIndex for collectionmanager
Statistics.java : added updateAchievementsArray

SEOKMINJUN committed:
Merge pull request #24 from hy-clove/KAN-40/Fix-Sound

KAN-40/Fix sound is not playing on exe

SEOKMINJUN commented:
Please check there is any critical bug before release

SEOKMINJUN committed:
Fix : Duplicate achievements are created when call getAchievementIndex that implemented at CollectionManager.

SEOKMINJUN committed:
Fix : Achievement not draw when acquire, Fix logic when acquire achievement.

RyuJeMu committed:
Resolve irregular distance-calculation on double-tab

RyuJeMu committed:
Include cumulative distance variable

RyuJeMu committed:
Display total distance through scree readjustment

RyuJeMu committed:
Manage the totalDistance variable within th GameScreen

RyuJeMu reviewed (approved):
Cannot find related critical bugs

SEOKMINJUN committed:
Merge pull request #25 from hy-clove/KAN-39-Fix-Calculating-distance-abnormal-functionality

SEOKMINJUN committed:
Fix : Remove bomb item probability set to 100%

SEOKMINJUN committed:
Fix : Not save coin and gem amount to currency file

SEOKMINJUN committed:
Fix : When clear wave, screen is not cleared. Remove some entities
- Item
- Bullet
- Obstacle

DongjunSuh923 committed:
-Added new page for achievement collections for distance achievements
-Added new achievement set for distance achievements

DongjunSuh923 committed:
Merge remote-tracking branch 'origin/develop' into KAN-10-Fix-Achievements-Achievement-System-not-working

# Conflicts:
#	src/screen/GameScreen.java

SEOKMINJUN committed:
Merge pull request #27 from hy-clove/KAN-10-Fix-Achievements-Achievement-System-not-working

KAN-10 fix achievements achievement system not working

woojinss committed:
Create : Screen Test File
Created ScreenTest File For Test Screens

woojinss committed:
Modified : CollectionsScreenCode Variable
Instead of using CollectionsScreenCode as a static variable, Lombok is used, and its location is moved to CollectionsManager.

HanRH99 committed:
Feat/Integrate highscore and recentscore menu

HanRH99 committed:
Feat/Integrate highscore and recentscore menu

RyuJeMu committed:
Add Distance-Calculation-Test system

DongjunSuh923 committed:
.

DongjunSuh923 committed:
-Added Junit Test against Achievement and AchievementManager

RyuJeMu committed:
Add a test and comments

DongjunSuh923 committed:
-Add one more test to AchievementManagerTest

RyuJeMu committed:
Add Double-Tab-Test system

SEOKMINJUN committed:
Merge pull request #28 from hy-clove/KAN-48-Feat-Integrate-highscore-and-recentscore-menu

Feat/Integrate highscore and recentscore menu

SEOKMINJUN committed:
Merge pull request #29 from hy-clove/KAN-49-Feat-Junit-Test-Achievements

Kan 49 feat junit test achievements

woojinss committed:
Modified : CollectionManager Test
It was determined that testing the CollectionManager class would be more beneficial than focusing on the Collection Screen for now. Therefore, the existing file has been deleted, and a new CollectionManagerTest class has been created.

woojinss committed:
Modified : Adjusted Position of Achievement Clear Pop-up
The current achievement clear window was being obstructed by other UI elements, so its position has been adjusted.

DongjunSuh923 committed:
-Statistics.java : Added achievements to scheduler, Now the achievements except the distance achievement(It works quite different compared to other achievements) works perfectly

SEOKMINJUN committed:
Fix : Game not end if ship destroy with more than one wave remaining

SEOKMINJUN committed:
KAN-44 Fix : Game not end if ship destroy with more than one wave remaining on 2 player mod

SEOKMINJUN committed:
Chore: Add test step at build workflow

SEOKMINJUN committed:
Chore: Upload report when test failed on build workflow

SEOKMINJUN committed:
Chore: Change report's artifact id

SEOKMINJUN committed:
Fix : Add exception handle on loading mp3 when running test on ubuntu environment(For github action)

woojinss committed:
Fix : Show Achievement Clear Condition
If an achievement has ever been cleared at least once, its condition will be made visible.

DongjunSuh923 committed:
-Fixed wrong conditions for distance achievement
-Added distance strings to FileManager
-GameScreen now updates totalDistance to Statistics
-Added distance to score class
-Receives total distance from ScoreScreen
-Added method comDistance to Statistics and scheduler

woojinss committed:
Fix : Move to Flie

DongjunSuh923 committed:
.

DongjunSuh923 committed:
- Collection tab now prints achievement 'completed' instead of writing '1'

DongjunSuh923 committed:
- Now the Collections tab will print '---' instead of 0 & 'COMPLETED' instead of 1 in achievement page
- The written "???" under the name of achievements at the achievement page(Collections) will now print the descriptions about the according achievement

DongjunSuh923 committed:
Merge pull request #32 from hy-clove/KAN-42-Fix-Achievements-Add-achievements-to-the-scheduler-Updates-every-time

-Statistics.java : Added achievements to scheduler, Now the achieveme…

DongjunSuh923 committed:
Merge remote-tracking branch 'origin/develop' into KAN-56-Fix-Add-Distance-to-statistics-make-the-achievement-functional

# Conflicts:
#	src/engine/Achievement/AchievementManager.java

DongjunSuh923 committed:
Reset Collections.properties

SEOKMINJUN committed:
Merge pull request #30 from hy-clove/KAN-37-Feat-JUnit-Test-Double-tab-Distance

Kan 37 feat j unit test double tab distance

SEOKMINJUN committed:
Merge pull request #33 from hy-clove/KAN-45-Feat-JUnit-Test-Create-a-compendium-tab-to-record-encountered-enemies,-achievements,-and-more

Kan 45 feat j unit test create a compendium tab to record encountered enemies, achievements, and more

SEOKMINJUN committed:
Merge branch `develop` into KAN-56-Fix-Add-Distance-to-statistics-make-the-achievement-functional

SEOKMINJUN committed:
Merge pull request #34 from hy-clove/KAN-56-Fix-Add-Distance-to-statistics-make-the-achievement-functional

Kan 56 fix add distance to statistics make the achievement functional & fix collection tab to print Descriptions and 'COMPLETED' at achievements page

woojinss committed:
Merge branch 'develop' into KAN-44-Integrate-Achievement-with-Collection-Screen

# Conflicts:
#	src/engine/DrawManager.java

woojinss committed:
Merge : Merging with the develop branch

HanRH99 committed:
FileManager JUnit test

woojinss committed:
Fix : After Merging the Branch, Modifications Related to Achievements in the Collection
Clean Up the Messy Code After Merging the Branch

woojinss committed:
Modified : Providing Visual Effect for Achievement Completion
To make the achievement completion more apparent, a visual effect is provided by changing the color of the achievement name when completed

woojinss committed:
Fix : A Few Modifications
Divided into 'complete' and 'incomplete' to display them separately, along with some other minor code fixes.

SEOKMINJUN committed:
Merge pull request #37 from hy-clove/KAN-44-Integrate-Achievement-with-Collection-Screen

Kan 44 integrate achievement with collection screen

SEOKMINJUN committed:
Merge pull request #31 from hy-clove/KAN-54/Chore-RunTestWorkflow

KAN-54/Chore:Add test step at build workflow

SEOKMINJUN committed:
Merge pull request #35 from hy-clove/KAN-57-Feat-JUnit-Test-FileManager

FileManager JUnit test

HanRH99 committed:
KAN 53 JUnit Test Score

SEOKMINJUN committed:
Merge pull request #26 from hy-clove/KAN-41/Fix-Several

KAN-41/Fix several things

SEOKMINJUN committed:
Merge pull request #38 from hy-clove/KAN-53-Feat-JUnit-Test-Score

KAN 53 JUnit Test Score

SEOKMINJUN committed:
Fix : Fix test failed on testCheckAchievementUnlock.
- After when we changed achievement system integrate to Statistic class. Achievement check Statistics value instead input value

SEOKMINJUN committed:
Merge pull request #39 from hy-clove/KAN-41/Fix-Several

KAN-41 Fix achievement test
